### PR TITLE
Add BUILD_TOOLS compile flag

### PR DESCRIPTION
**Pull request by TheGreatRambler**
Adds a `BUILD_TOOLS` compile flag so that compiling the library only is possible. Saves both time and storage in cases when the library is embedded in another program. Is set by default

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -270,7 +273,9 @@ else()
   if(WIN32)
     add_compile_flag("-D_GNU_SOURCE")
     add_compile_flag("-D__STDC_FORMAT_MACROS")
-    add_link_flag("-Wl,--stack,8388608")
+    if(NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+      add_link_flag("-Wl,--stack,8388608")
+    endif()
```
Comment (kripken) [inline CMakeLists.txt:278]: This change looks unrelated?

**Review by kripken — COMMENTED** [review]
Looks like a reasonable option to add.

Curious if people have thoughts on the name of the new option? cc @sbc100

File: CMakeLists.txt [hunk 2]
```diff
@@ -37,6 +37,9 @@ option(BUILD_TESTS "Build GTest-based tests" ON)
 # Turn this on to build binaryen.js as ES5, with additional compatibility configuration for js_of_ocaml.
 option(JS_OF_OCAML "Build binaryen.js for js_of_ocaml" OFF)
 
+# Turn this off to build only the library
+option(BUILD_TOOLS "Build tools" ON)
```
Comment (sbc100) [inline CMakeLists.txt:41]: Maybe group this with `BUILD_TESTS` above?

**Review by sbc100 — COMMENTED** [review]
Seems reasonable to me.

File: CMakeLists.txt (hunk 1 above)
Comment (TheGreatRambler) [reply CMakeLists.txt:278]: Ah my bad, should have made a separate PR for this. `--stack` doesn't exist as a linker option on clang, and because clang in mingw64 isn't considered MSVC, this flag is used and causes errors on linking

Comment (TheGreatRambler) [reply CMakeLists.txt:280]: Pushed a new commit to add back the stack, using `-stack_size` which clang supports

Comment (sbc100) [reply CMakeLists.txt:280]: Can you make this a completely separate PR since it seems unrelated?

Also, I think maybe using `if(MSVC)` is better than checking or `NOT Clang` here?

Comment (TheGreatRambler) [reply CMakeLists.txt:280]: `MSVC` is not set but clang is in my case. Perhaps that's an edge case not worth supporting but I'm using a standard clang install with the msys shell which works for my other libs. Also yeah I'll split that into another PR, how can I do that

**Comment by sbc100** [comment]
I still think the BUILD_TOOLS change is worth landing.
